### DISCUSSION TITLE
Incline help: AppleTV (Zwift) <----> iOS (Qdomyos) <---> raspberry pi on treadmill (Qdomyos)

### DISCUSSION THREAD

**Discussion by skoregon** [cagnulein/qdomyos-zwift#2318] [Q&A]
I see the toggles for incline control in the iOS version of Qdomyos as well as the login for zwift.  I have an older version of qdomyos setup on a raspberry pi to control my treadmill via the GPIO.  The pi works fine when connected to Zwift but no incline data during running.  I saw in TestFlight it says you can get incline data when running on iOS.   How do I do that and get it sent to Qdomyos on my treadmill?  If I use the workout hack directly w/ the qdomyos on the pi it controls my incline on treadmill so I know the hardware is correct.

**Comment by cagnulein** [comment]
Hi @skoregon i didn't port the zwift hack on linux yet. i will give it a try in the next weeks

**Reply by skoregon** [reply]
Thank you Roberto, you are so helpful.   How do I know if the iOS version of Qdomyos is receiving (decoding?) incline data during a running session?

**Reply by cagnulein** [reply]
If you see the green check Mark on the zwift username in qz you are good to go!

**Reply by skoregon** [reply]
Thank you Roberto.  It would be awesome if the zwift back was also in the Linux version.

**Reply by cagnulein** [reply]
Thanks for the reminder, in these weeks I was very busy, keep me pinged!

**Comment by skoregon** [comment]
[debug-Tue_Aug_6_20_07_29_2024.log](https://github.com/user-attachments/files/16517850/debug-Tue_Aug_6_20_07_29_2024.log)
[debug-Tue_Aug_6_20_06_50_2024.log](https://github.com/user-attachments/files/16517851/debug-Tue_Aug_6_20_06_50_2024.log)

Hi Roberto!

I attached two debug files running Qdomyos on my iPhone, DomyosBridge on my raspberry pi and Zwift on appletv. The Qdomyos part seems good in that the speed is transmitted from the DomyosBridge to Qdomyos to Zwift.  The incline appears to be received from Zwift to Qdomyos on the phone but doesn't appear to send to DomesBridge on the pi.  That is, the T.incline (%) tile matches Zwift but the "Inclination" tile remains at zero.  I am attempting to use the Zwift login route to get incline.  The incline worked when it was Zwift (AppleTV)->DomyosBridge on Raspberry pi using the older meetup hack.

**Comment by cagnulein** [comment]
hi a debug log from the raspberry is needed too.
Thanks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 7 ago 2024 alle ore 05:33 skoregon ***@***.***>
ha scritto:

**Comment by skoregon** [comment]
[debug-Tue_Aug_6_20_36_16_2024.log](https://github.com/user-attachments/files/16555944/debug-Tue_Aug_6_20_36_16_2024.log)
I apologize for how large they are.  verbose indeed.
[debug-Tue_Aug_6_13_24_51_2024.log](https://github.com/user-attachments/files/16555957/debug-Tue_Aug_6_13_24_51_2024.log)

**Comment by cagnulein** [comment]
@skoregon in this log the incline is always 0 except for a very tiny moment here

```
Tue Aug 6 20:08:18 2024 1723000098759 Debug: gpiotreadmill.cpp void gpiotreadmill::forceIncline(double) "gpiotreadmill.cpp: request set Incline 0 to 0.5"
Tue Aug 6 20:08:18 2024 1723000098775 Debug: gpiotreadmill.cpp virtual void gpioWorkerThread::run() "increasing incline from 0 to 0.5"
Tue Aug 6 20:08:20 2024 1723000100233 Debug: gpiotreadmill.cpp void gpiotreadmill::forceIncline(double) "gpiotreadmill.cpp: request set Incline 0.5 to 0"
Tue Aug 6 20:08:20 2024 1723000100246 Debug: gpiotreadmill.cpp virtual void gpioWorkerThread::run() "decreasing incline from 0.5 to 0"
```

**Reply by skoregon** [reply]
I believe that may be when I tapped the control to increase it half a step to make sure it worked.  I will record another set of logs where I am more methodical.  The debug logs should have been on the “oh hill no” Zwift map which is just a giant uphill track.

**Comment by cagnulein** [comment]
ok give it another go, unfortunately this was not included it :(

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 10 ago 2024 alle 02:08 skoregon ***@***.***> ha
scritto:
